src/main/java/teammates/logic/backdoor/BackDoorLogic.java

```diff
@@ -439,8 +439,13 @@ public void removeDataBundle(DataBundle dataBundle) {
         }
         accountsDb.deleteAccounts(dataBundle.accounts.values());
 
-        // Delete all admin emails
-        adminEmailsDb.deleteEntities(adminEmailsDb.getAllAdminEmails());
+        for (AdminEmailAttributes email : dataBundle.adminEmails.values()) {
+            AdminEmailAttributes emailInDb = adminEmailsDb.getAdminEmailBySubject(email.subject);
+            if (emailInDb == null) {
```

Comment:
1. Usually when preparing test data, a remove operation is done first followed by the persist operation, so the test data might not have existed and can be null. The deleting for other entities will also fail silently if they cannot find the entities corresponding to items in the list. The reason why only AdminEmailAttributes needs to handle the null value here is because the usual keys for the entity (emailId or (subject, createDate) ) cannot be specified through DataBundle.